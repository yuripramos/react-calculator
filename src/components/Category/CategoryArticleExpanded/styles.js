import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { grey10, red10, darkestBlue } from "../../../styles/settings";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;

export const CategoryWrapper = styled.div`
  color: #000;
  font-family: "CrimsomText Regular";
  text-align: left;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(100)};
  display: inline-flex;
  flex-wrap: wrap;
  border-bottom: solid 1px ${grey10};
  :nth-child(2n) {
    background: rgba(229, 229, 232, 0.1);
  }
`;

export const Link = styled.a`
  text-decoration: none;
  font-family: "OpenSans Regular";
  font-size: ${rem(16)};
  color: ${darkestBlue};
  cursor: pointer;
`;

export const Media = styled.img`
  width: 100%;
  height: ${rem(350)};
    ${media.md(css`
    height: ${rem(500)};
  `)};
  ${media.lg(css`
    height: ${rem(620)};
  `)};
`;

export const Title = styled.span`
  font-size: ${rem(38)};
  font-weight: 200;
  line-height: 0.75;
  font-family: "CrimsomText SemiBold";
  margin-top: ${rem(30)};
  width: 100%;
`;

export const PublishedDate = styled.span`
  font-family: "OpenSans Regular";
  font-size: ${rem(16)};
  margin: 1.625rem 0;
  letter-spacing: 0.05em;
  color: ${red10};
  width: 100%;
  margin: ${rem(20)} 0 ${rem(15)} 0;
`;

export const Author = styled.span`
  font-family: "OpenSans Regular";
  font-size: ${rem(12)};
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;

export const Kicker = styled.span`
  font-size: ${rem(18)};
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;

export const Description = styled.span`
  font-size: ${rem(22)};
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;


