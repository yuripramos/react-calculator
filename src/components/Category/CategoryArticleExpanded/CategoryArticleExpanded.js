import React, { Component } from "react";
import moment from "moment";
import { MODAL_TYPES } from "../../common/Modal/Modal";
import { hardRedirect } from "../../../utils/redirect";
import { Container, Row, Column } from "../../../styles/grid";
import { string, arrayOf, shape } from "prop-types";

import {
  ContentWrapper,
  Title,
  CategoryWrapper,
  PublishedDate,
  Author,
  Media,
  Kicker,
  Link,
  Description
} from "./styles";

class CategoryContentExpanded extends Component {
  constructor(props) {
    super(props);
    this.state = {
      content: false,
    }
    this.handleConfirmButton = this.handleConfirmButton.bind(this);
  }
  async handleConfirmButton(url) {
    const { openModal, closeModal } = this.props;
    openModal({
      title: "You are leaving this domain.",
      type: MODAL_TYPES.DEFAULT,
      icon: "Attention",
      description: ["We hope you enjoyed our project =)"],
      onClose: () => {
        hardRedirect(url);
        return closeModal()
      }
    });
  }
  componentDidMount() {
    const { current, match } = this.props;
    if(current.status) {
      this.setState({
        content: current.results.find(
          e => {
            if (e.short_url) {
              return e.short_url.includes(match.params.idArticle);
            }
            return e.url.includes(match.params.idArticle);
          }
        ),
      })
    }
  }

  render() {
    const { match } = this.props;
    const { content } = this.state;
    return (
      <ContentWrapper>
        {content && (
          <Container>
            <Row>
              <Column>
                <CategoryWrapper key={`article-${match.params.idArticle}`}>
                  <Title>{content.title} </Title>
                  {content.kicker && <Kicker>{content.kicker}</Kicker>}
                  <PublishedDate>
                    updated in:{" "}
                    {moment(content.published_date).format("LLL")}
                  </PublishedDate>
                  {content.multimedia.length > 0 && (
                    <Media
                      src={
                        content.multimedia[4].url || content.multimedia[2].url
                      }
                      alt="thumbnail"
                    />
                  )}
                  <Description>
                    {content.abstract}
                  </Description>
                  <Author>{content.byline} </Author>
                  <Link onClick={() => this.handleConfirmButton(content.url)}>Original Link from NY Times</Link>
                </CategoryWrapper>
              </Column>
            </Row>
          </Container>
        )}
      </ContentWrapper>
    );
  }
}

CategoryContentExpanded.propTypes = {
  current: shape({
      results: arrayOf(
        shape({
          url: string,
          short_url:string,
        })
      )
    }),
  content: arrayOf(
    shape({
      title: string,
      published_date: string,
      byline: string
    })
  ),
  match:
    shape({
      path: string,
    }),
};

export default CategoryContentExpanded;
