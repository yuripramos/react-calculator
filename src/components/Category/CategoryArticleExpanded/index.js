import CategoryArticleExpanded from "./CategoryArticleExpanded";
import { withRouter } from "react-router-dom";
import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";

import modalActions from "../../../actions/modal";
import newsActions from "../../../actions/News";

export default connect(
  ({ topUserStories, isFilled, FeedArray, current }) => ({
    topUserStories,
    isFilled,
    FeedArray,
    current
  }),
  combineActions(newsActions, modalActions)
)(withRouter(CategoryArticleExpanded));
