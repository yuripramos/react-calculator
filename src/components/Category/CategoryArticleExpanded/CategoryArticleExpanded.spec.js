import React from "react";
import { shallow, mount } from "enzyme";

import { spy } from "sinon";
import CategoryArticleExpanded from "./CategoryArticleExpanded";

let props;

beforeEach(() => {
  props = {
    current: [
      {
        section: "science",
        abstract: "while ultraviolete fluor",
        url: "http://www.nytimes.com/2019/02/01/science/pink-glow",
        published_date: "2019-02-01T12:42:18-05:00",
        byline: "By VERONIQUE GREENWOOD",
        title: "Flying Squirrels That Glow Pink in the Dark",
        multimedia: [
          { url: "www.imgfor.be/23213.jpg" },
          { url: "www.imgfor.be/932sdas.jpg" },
          { url: "www.imgfor.be/3JWM3.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" }
        ]
      },
      {
        section: "science",
        abstract: "while usadasltraviolete fluor",
        url: "http://www.nytimes.com/2019/02/01/science/red-glow",
        published_date: "2019-02-01T12:42:16-05:00",
        byline: "By BARBARIAN",
        title: "Pink is the new white",
        multimedia: [
          { url: "www.imgfor.be/23213.jpg" },
          { url: "www.imgfor.be/932sdas.jpg" },
          { url: "www.imgfor.be/3JWM3.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" }
        ]
      }
    ],
    isMainPage: false,
    match: {
      path: "/science/hot",
      params: {
        idArticle: "EfS2s"
      }
    },
    openModal: jest.fn()
  };
});

const contentState = {
  section: "science",
  abstract: "while usadasltraviolete fluor",
  url: "http://www.nytimes.com/2019/02/01/science/red-glow",
  published_date: "2019-02-01T12:42:16-05:00",
  byline: "By BARBARIAN",
  title: "Pink is the new white",
  multimedia: [
    { url: "www.imgfor.be/23213.jpg" },
    { url: "www.imgfor.be/932sdas.jpg" },
    { url: "www.imgfor.be/3JWM3.jpg" },
    { url: "www.imgfor.be/9FSww.jpg" },
    { url: "www.imgfor.be/9FSww.jpg" }
  ]
};
describe("CategoryArticleExpanded Component", () => {
  it("should match snapshot with props", () => {
    expect(
      shallow(<CategoryArticleExpanded {...props} />)
    ).toMatchSnapshot();
  });

  it("should setState to apply the content", () => {
    spy(CategoryArticleExpanded.prototype, "componentDidMount");

    const wrapper = mount(<CategoryArticleExpanded {...props} />);
    wrapper.setState({ content: contentState }, () => {
      wrapper.update();
      expect(
        CategoryArticleExpanded.prototype.componentDidMount.calledOnce
      );
      expect(wrapper.state("content")).toEqual(contentState);
    });

    CategoryArticleExpanded.prototype.componentDidMount.restore();
  });
});
