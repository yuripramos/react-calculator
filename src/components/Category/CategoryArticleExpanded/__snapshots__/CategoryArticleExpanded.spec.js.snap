// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryArticleExpanded Component should match snapshot with props 1`] = `<styled.div />`;
